### DISCUSSION TITLE
Working nginx reverse proxy example config

### DISCUSSION THREAD

**Discussion by Sieboldianus** [photoview/photoview#449] [Show and tell]
I thought I share my working nginx config for photoview docker on a local port. I had troubles with Web Socket connections
("wss: Error during WebSocket handshake: Unexpected response code: 200"), but could fix it by including:
```nginx
    proxy_set_header Upgrade $http_upgrade;
    proxy_set_header Connection "upgrade";
```

Here is the full config:
```nginx
server {
    listen 80;
    listen [::]:80;
    listen 443 ssl http2;
    listen [::]:443 ssl http2;

    server_name photo.local.mytld.com;

    access_log off;
    error_log /var/log/nginx/error.log crit;

    ssl_certificate    /etc/nginx/ssl/wildcard.local.mytld.com.fullchain;
    ssl_certificate_key /etc/nginx/ssl/wildcard.local.mytld.com.key;

    # TLS
    ssl_protocols TLSv1.2;
    ssl_ciphers HIGH:!MEDIUM:!LOW:!aNULL:!NULL:!SHA;
    ssl_prefer_server_ciphers on;
    ssl_session_cache shared:SSL:10m;

    location / {
        proxy_pass http://127.0.0.1:8001/;
        proxy_set_header X-Forwarded-For $remote_addr;
        proxy_set_header Host $host;
        proxy_http_version 1.1;    # to keep alive
        proxy_set_header Upgrade $http_upgrade;
        proxy_set_header Connection "upgrade";
    }

    # redirect to https
    if ($https = '') { return 301 https://$host$request_uri; }
}

```

**Comment by hunkyn** [comment]
Thank you for this. I am using the linuxserver SWAG container. Can i just copy this and modify and place it under "proxy-confs" folder? or is there anything else I need to do?

**Comment by Sieboldianus** [comment]
I don't know SWAG, but looking at [their docs](https://docs.linuxserver.io/general/swag#preset-proxy-confs), it sounds like you can just copy the above config into that folder, yes. SWAG seems to do the ssl_certificate stuff for you, so check out their other preconfigured confs and replace these two lines from my config with theirs:

```conf
ssl_certificate    /etc/nginx/ssl/wildcard.local.mytld.com.fullchain;
ssl_certificate_key /etc/nginx/ssl/wildcard.local.mytld.com.key;
```

**Reply by hunkyn** [reply]
Thanks ended up copying their template and just updated url and port and it works great.

**Reply by ghost** [reply]
@hunkyn  would you mind sharing? I just did that but it doesn't do anything when trying to log in.
